components/Input/Input.tsx

```diff
@@ -0,0 +1,59 @@
+import { forwardRef } from 'react';
+import { VariantProps } from 'class-variance-authority';
+import { FieldError, useFormContext } from 'react-hook-form';
+import InputClasses from './Input.classes';
+
+interface InputBaseProps extends VariantProps<typeof InputClasses> {
+  label?: string;
+  name: string;
+}
+
+// TODO: Icon will be displayed in error situations
+type Props = React.InputHTMLAttributes<HTMLInputElement> & InputBaseProps;
+
+/**
+ * @example <Input name="email" label="Email" placeholder="olivia@untitledui.com"  />
+ *  */
+const Input = forwardRef<HTMLInputElement, Props>(
+  ({ className, variant, label, size, ...rest }, ref) => {
+    const {
+      formState: { errors }
+    } = useFormContext();
+
+    return (
+      <div>
+        {label && (
+          <label className="block font-medium text-sm mb-1 text-gray-800">
+            {label}
+          </label>
+        )}
+        <input
+          className={InputClasses({
+            variant,
+            size,
+            error: !!errors[rest.name]
+          })}
+          ref={ref}
+          {...rest}
+        />
+        <ErrorField name={rest.name} />
```

Comment:
Rest is not clear enough to understand what the prop works for. it should be like errorMessage="bla bla bla"
+ you can add onChange prop and can remove spread props